### PR TITLE
Move TATF based tests to MSTest

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -65,12 +66,13 @@
...
     <PackageVersion Include="Microsoft.Extensions.FileSystemGlobbing" Version="9.0.0" />
-    <PackageVersion Include="Microsoft.Testing.Internal.Framework" Version="$(MicrosoftTestingInternalFrameworkVersion)" />
-    <PackageVersion Include="Microsoft.Testing.Internal.Framework.SourceGeneration" Version="$(MicrosoftTestingInternalFrameworkVersion)" />
     <PackageVersion Include="Microsoft.TestPlatform" Version="$(MicrosoftNETTestSdkVersion)" />
     <!-- Pinned to 4.18.4 for security -->
     <PackageVersion Include="Moq" Version="4.18.4" />
     <PackageVersion Include="MSBuild.StructuredLogger" Version="2.2.386" />
+    <PackageVersion Include="MSTest.Analyzers" Version="$(MSTestVersion)" />
+    <PackageVersion Include="MSTest.TestAdapter" Version="$(MSTestVersion)" />
+    <PackageVersion Include="MSTest.TestFramework" Version="$(MSTestVersion)" />
```
Comment (Youssef1313) [inline Directory.Packages.props:75]: Could we switch to the meta package?

File: samples/Playground/Tests.cs [hunk 2]
```diff
@@ -18,7 +18,7 @@ public class TestClass
     public void Test() => TestContext.AddResultFile(@"c:\hello2");
 
     [TestMethod]
-    public void Test2() => Assert.AreEqual(1, 0, "few");
+    public void Test2() => Assert.AreEqual(0, 1, "few");
```
Comment (Youssef1313) [inline samples/Playground/Tests.cs]: Unnecessary change but it doesn't harm.

File: src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs [hunk 3]
```diff
@@ -195,14 +195,15 @@ private void AddProperty(string name, string value)
         {
             if (!_filterDictionaryBuilder.TryGetValue(name, out ImmutableHashSet<string>.Builder? values))
             {
-                values = ImmutableHashSet.CreateBuilder<string>(StringComparer.OrdinalIgnoreCase);
+                values = ImmutableHashSet.CreateBuilder(StringComparer.OrdinalIgnoreCase)!;
```
Comment (Youssef1313) [inline src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs:198]: Why is the suppression needed?

File: test/IntegrationTests/Microsoft.Testing.Platform.Acceptance.IntegrationTests/Properties/launchSettings.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "profiles": {
     "Microsoft.Testing.Platform.Acceptance.IntegrationTests": {
       "commandName": "Project",
-      "commandLineArgs": "--treenode-filter /*/*/*/**",
+      // "commandLineArgs": "--treenode-filter /*/*/*/**",
```
Comment (Youssef1313) [inline test/IntegrationTests/Microsoft.Testing.Platform.Acceptance.IntegrationTests/Properties/launchSettings.json]: Is this change intended?

File: Directory.Packages.props (hunk 1 above)
Comment (Evangelink) [reply Directory.Packages.props:75]: I tried and something wasn't working but I don't recall what. Let me double check.

File: samples/Playground/Tests.cs (hunk 2 above)
Comment (Evangelink) [reply samples/Playground/Tests.cs]: Part of the bulk fix on other assertions, I'll leave it as-is.

File: test/IntegrationTests/Microsoft.Testing.Platform.Acceptance.IntegrationTests/Properties/launchSettings.json (hunk 4 above)
Comment (Evangelink) [reply test/IntegrationTests/Microsoft.Testing.Platform.Acceptance.IntegrationTests/Properties/launchSettings.json]: Yes! The filter isn't supported by MSTest so when you try to run the project it fails. I need to double check this is done everywhere!

File: test/UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests.csproj [hunk 5]
```diff
@@ -1,9 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>$(MicrosoftTestingTargetFrameworks);net462</TargetFrameworks>
+    <TargetFrameworks>net8.0;net9.0</TargetFrameworks>
```
Comment (Youssef1313) [inline test/UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests.csproj:4]: Maybe use `$(NetMinimum);$(NetCurrent)`? (I assume NetMinimum flows properly from Arcade.

File: test/Utilities/Microsoft.Testing.TestInfrastructure/Microsoft.Testing.TestInfrastructure.csproj [hunk 6]
```diff
@@ -7,18 +7,17 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Testing.Internal.Framework" />
+    <PackageReference Include="MSTest.TestFramework" />
```
Comment (Youssef1313) [inline test/Utilities/Microsoft.Testing.TestInfrastructure/Microsoft.Testing.TestInfrastructure.csproj:11]: Should we explicitly reference MSTest.Analyzers here as well? At least until we are using 3.8 which will have https://github.com/microsoft/testfx/pull/4351

File: Directory.Packages.props (hunk 1 above)
Comment (Evangelink) [reply Directory.Packages.props:75]: Looks like it was just to avoid `Microsoft.NET.Test.Sdk` (the VSTest deps) that is part of the bundle.

File: src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs (hunk 3 above)
Comment (Evangelink) [reply src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs:198]: It's complaining about inconsistent nullability.

File: test/Utilities/Microsoft.Testing.TestInfrastructure/Microsoft.Testing.TestInfrastructure.csproj (hunk 6 above)
Comment (Evangelink) [reply test/Utilities/Microsoft.Testing.TestInfrastructure/Microsoft.Testing.TestInfrastructure.csproj:11]: Yes!

File: test/UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests.csproj (hunk 5 above)
Comment (Evangelink) [reply test/UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests/Microsoft.Testing.Platform.MSBuild.UnitTests.csproj:4]: It would not work well when we move to arcade 10, we would only test `net8.0` and `net10.0`

File: src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs (hunk 3 above)
Comment (Youssef1313) [reply src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/FastFilter.cs:198]: Oh, was the code previously creating `ImmutableHashSet<string>` and it's not `ImmutableHashSet<string?>` ? In that case, I think it's better to keep the type parameter explicitly specified as `string`

**Review by MarcoRossignoli — APPROVED** [review]
